Getting In-Car Audio updates 4.7 updates (#127)
smartdevicelink/sdl_android_guides: docs/Migrating to Newer SDL Android Versions/Updating to 4.7/index.md

After removing `IProxyListenerALM ` from the `SdlService`, all of its previously overridden functions will need to be removed. If your app used any of these callback methods, it will help to document which ones they were, as you will need to add in the listeners that you need using the `SdlManager`'s `addOnRPCNotificationListener`.

!!! note

> When you start using the managers, you have to make sure that your app subscribes to notifications before sending the corresponding RPC requests or else some notifications may be missed.

Reviewer: ```suggestion
When you start using the managers, you have to make sure that your app subscribes to notifications before sending the corresponding RPC requests and subscriptions or else some notifications may be missed.
```

	//...
});
```

### Subscribing to AudioPassThru Notifications

> Previously, your `SdlService` had to implement `IProxyListenerALM` interface which means your `SdlService` class had to override all of the `IProxyListenerALM` callback methods including `onOnAudioPassThru`.

Reviewer: ```suggestion

Previously, your `SdlService` had to implement `IProxyListenerALM` interface which means your `SdlService` class had to override all of the `IProxyListenerALM` callback methods including `onOnAudioPassThru`.
```

```java
@Override
public void onOnAudioPassThru(OnAudioPassThru notification) {
...
}
```

> In 4.7, to subscribe to a notification, your app has to call `addOnRPCNotificationListener` to add a listener for the notification that your app wants to subscribe to. The listener should be added before sending RPC request or else some notifications may be missed. 

Reviewer: ```suggestion
In 4.7 and the new manager APIs, in order to receive the  `OnAudioPassThru ` notifications, your app must add a `OnRPCNotificationListener` using the `SdlManager`'s method `addOnRPCNotificationListener`. This will subscribe the app to any notifications of the provided type, in this case `ON_AUDIO_PASS_THRU`. The listener should be added before sending the corresponding RPC request/subscription or else some notifications may be missed. 
```

```java
sdlManager.addOnRPCNotificationListener(FunctionID.ON_AUDIO_PASS_THRU, new OnRPCNotificationListener() {
    @Override
    public void onNotified(RPCNotification notification) {
        ...
    }
});

.
.

> 

Reviewer: Please add the APT example code here
